vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -490,6 +491,11 @@ def initialize_dummy_weights(
     """
     for param in model.state_dict().values():
         if torch.is_floating_point(param):
+            if current_platform.is_tpu():
+                # XLA device does not support torch.Generator()
+                param.uniform_(low, high)
```

Comment:
how does xla support seed then?

Reply:
They have an API, which we use in `init_device`: https://github.com/vllm-project/vllm/blob/533d1932d23917f7a29ba04c6be5c8b2676b6969/vllm/worker/tpu_worker.py#L97

Reply:
Does it support per-function seed?

Reply:
Not at the moment according to the pytorch xla team.